lib/components/viewers/pattern-row.js

```diff
@@ -0,0 +1,210 @@
+import React, { Component } from 'react'
```

Comment:
Note for reviewer: This component is mostly extracted from stop-viewer to reduce that file's size.